src/generators/dockerHelper.js

```diff
@@ -50,39 +53,79 @@ export function buildDockerCompose(config) {
     'lib/docker-compose-definitions-cakeshop.yml',
   )) : ''
 
+  const splunkDefinitions = hasSplunk ? readFileToString(joinPath(
+    libRootDir(),
+    'lib/docker-compose-definitions-splunk-helpers.yml'
+  )) : ''
+
   let services = config.nodes.map((node, i) => {
-    let allServices = buildNodeService(config, node, i, hasTessera)
+    let allServices = buildNodeService(config, node, i, hasTessera, hasSplunk, txGenerate)
     if (hasTessera) {
-      allServices = [allServices, buildTesseraService(config, node, i)].join('')
+      allServices = [allServices, buildTesseraService(config, node, i, hasSplunk)].join('')
     }
     return allServices
   })
   if (hasCakeshop) {
-    services = [services.join(''), buildCakeshopService(config)]
+    services = [services.join(''), buildCakeshopService(config, hasSplunk)]
+  }
+  if (hasSplunk) {
+    services = [services.join(''),
+      buildEthloggerService(config),
+      buildCadvisorService(config)]
+  }
+  if (txGenerate) {
+    let pubkeys = []
+    config.nodes.forEach((_, i) => {
+      pubkeys.push(loadTesseraPublicKey(config, i + 1))
+    });
```

Comment:
this fails if the user chooses 'none' for the Tessera. You can check for isTessera(config.network.transactionManager).

But I'm guessing you'll need to selectively modify the txgenerator config to not try to send private transactions as well

Reply:
this was removed